### DISCUSSION TITLE
Chat permissions for moderators/admins

### DISCUSSION THREAD

**Discussion by sofy1801** [Zrips/CMI#3510] [Feature Requests]
Hi, I'm currently using a range of 500 blocks for chat messages but this way moderators and admins can't read chats unless they are near the player who sent the message. I haven't found a permission node that allows moderators to read all local chats so it would be nice if I could add it. 

Also this could be useful for chatrooms too. 
Thanks in advance

**Comment by CloudeLecaw** [comment]
you can add it

**Comment by mrfloris** [comment]
I do not know this works as I think it does, I haven't tested it; have you perhaps tried using the permission `cmi.chat.rangebypass - Allows to send messages over greater distances` Maybe this will not just let staff talk beyond the range, but also listen in?

**Comment by CloudeLecaw** [comment]
A good question. Just give it a try :)
If it does not work, then an extra permission for this would be nice

**Comment by sofy1801** [comment]
> I do not know this works as I think it does, I haven't tested it; have you perhaps tried using the permission `cmi.chat.rangebypass - Allows to send messages over greater distances` Maybe this will not just let staff talk beyond the range, but also listen in?

I've tried it already, with that moderators are able to read all but when they write in local chat the message will be global so it's not so great...

**Comment by Zrips** [comment]
@sofy1801 
Well in general you will ether send and see messages globally or only locally. Having them mixed up will introduce big confusions when you would see a message from some one asking for advice, you respond and then they ask again... Because you sent your message locally and they have been too far to see it. Unless i'm missing the point here or some one has some solution for it, i'm listening.

**Comment by sofy1801** [comment]
It's for surveillance purpose, I would like to be able to check that the players comply with the chat rules but if I have to just tell them to calm down I don't want the whole server to read my message without contest... I prefer to teleport in the zone and talk locally to the players that are involved.

**Comment by div2005** [comment]
> Well in general you will ether send and see messages globally or only locally. Having them mixed up will introduce big confusions when you would see a message from some one asking for advice, you respond and then they ask again... Because you sent your message locally and they have been too far to see it. Unless i'm missing the point here or some one has some solution for it, i'm listening.

It should look something like this:
**cmi.chat.spy** - permission which allows you to see messages regardless of distance. It should differ by some prefix
For example: **[SPY]**
Accordingly, a player with this perm Will receive something like this message:

`[SPY] [LocalChat] Player: Hi all!`

So, If I have GeneralRange: 100, and we are at a distance of more than 100 blocks from each other, only the one who has **cmi.chat.spy** should see the message of another player. And It is also important to note that a player with this permission remains able to write both in global chat and in local chat.

Also, check this: https://github.com/Zrips/CMI/issues/3260 and https://github.com/Zrips/CMI/issues/3444

**Comment by div2005** [comment]
Up it!

**Comment by alowave223** [comment]
will it come true?
